### PR TITLE
feat(eventing): add event specific details for io-engine stop event

### PR DESCRIPTION
**Pull request by datacore-vvarakantham**
Added event specific details for io-engine stop event.

Added error details to event specific details of nexus shutdown when the nexus shutdown fails.

Added subsystem pause/ resume and nexus init/reconfiguring events.

### CODE REVIEW

**Review by tiagolobocastro — APPROVED** [review]

**Review by sinhaashish — APPROVED** [review]

File: apis/events/protobuf/v1/event.proto [hunk 1]
```diff
@@ -55,6 +57,10 @@ enum EventAction {
   Shutdown = 18;
   Start = 19;
   Stop = 20;
+  SubsystemPause = 21;
```
Comment (tiagolobocastro) [inline apis/events/protobuf/v1/event.proto:60]: How is the initial pause event sent?

Comment (datacore-vvarakantham) [reply apis/events/protobuf/v1/event.proto:60]: I didn't add pause when we receive the pause request. I am creating a pause event when its completed. If there is error I am adding error to the event details. Else I am just adding duration for the pause action. So if there is no error it is assumed to be successful.

Comment (datacore-vvarakantham) [reply apis/events/protobuf/v1/event.proto:60]: Or should I add an event when pause request is received? And another event when the pause fails?

Comment (tiagolobocastro) [reply apis/events/protobuf/v1/event.proto:60]: Yes, 1 when it's started, and 1 when it's finished.
Reason for this is that pause sometimes gets stuck, in which case we'd never get the event, so would be good to know if a pause was started at some point or not.

Comment (datacore-vvarakantham) [reply apis/events/protobuf/v1/event.proto:60]: Added the events when subsystem pause started and completed.

**Comment by tiagolobocastro** [comment]
Can his be merged @datacore-vvarakantham ?

**Comment by datacore-vvarakantham** [comment]
> Can his be merged @datacore-vvarakantham ?

Yes, it can be merged.

**Pull request merged**